### PR TITLE
[Merged by Bors] - TO-3260 fix source preferences tests

### PR DESCRIPTION
**Pull request by rustonaut**
Turns all source preference tests into integration tests which work with storage. Most tests are just copied to the integration test file for source preferences and then slightly adapted.

**References:**

- [TO-3260]

**Based On:**

- #555 

[TO-3260]: https://xainag.atlassian.net/browse/TO-3260?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: discovery_engine/lib/src/discovery_engine_base.dart [hunk 1]
```diff
@@ -315,7 +315,7 @@ class DiscoveryEngine {
   /// duplicates found in provided sets, containing a set of said duplicates.
   /// - [EngineExceptionRaised] indicating a failed operation, with a reason
   /// for such failure.
-  Future<EngineEvent> overrideSources({
+  Future<EngineEvent> setSources({
```
[the remark this answers is not in the document]
Comment (rustonaut) [reply discovery_engine/lib/src/discovery_engine_base.dart:318]: let me undo this and add a `FIXME rename when we move to message passing API`.

**Comment by rustonaut** [comment]
ok, fixed test and added another test

**Review by janpetschexain — APPROVED** [review]
lgtm

**Comment by rustonaut** [comment]
bors merge

**Comment by rustonaut** [comment]
bors merge

**Pull request closed without merging**